jerry-core/ecma/builtin-objects/ecma-builtin-global.cpp

```diff
@@ -113,147 +113,157 @@ ecma_builtin_global_object_parse_int (ecma_value_t this_arg __attr_unused___, /*
-  bool strip_prefix = true;
-
-  /* 8. */
-  if (rad != 0)
-  {
-    /* 8.a */
-    if (rad < 2 || rad > 36)
+    /* 4. */
+    if (utf8_string_buff[start] == '-')
     {
-      ecma_number_t *ret_num_p = ecma_alloc_number ();
-      *ret_num_p = ecma_number_make_nan ();
-      ret_value = ecma_make_normal_completion_value (ecma_make_number_value (ret_num_p));
+      sign = -1;
     }
-    /* 8.b */
-    else if (rad != 16)
+
+    /* 5. */
+    if (utf8_string_buff[start] == '-' || utf8_string_buff[start] == '+')
     {
-      strip_prefix = false;
+      start++;
     }
-  }
-  /* 9. */
-  else
-  {
-    rad = 10;
-  }
 
-  if (ecma_is_completion_value_empty (ret_value))
-  {
-    /* 10. */
-    if (strip_prefix)
-    {
-      if (end - start >= 2
-          && utf8_string_buff[start] == '0'
-          && (utf8_string_buff[start + 1] == 'x' || utf8_string_buff[start + 1] == 'X'))
-      {
-        start += 2;
+    /* 6. */
+    ECMA_OP_TO_NUMBER_TRY_CATCH (radix_num, radix, ret_value);
+    int32_t rad = ecma_number_to_int32 (radix_num);
 
-        rad = 16;
-      }
-    }
+    /* 7.*/
+    bool strip_prefix = true;
 
-    /* 11. Check if characters are in [0, Radix - 1]. We also convert them to number values in the process. */
-    for (lit_utf8_size_t i = start; i < end; i++)
+    /* 8. */
+    if (rad != 0)
     {
-      if ((utf8_string_buff[i]) >= 'a' && utf8_string_buff[i] <= 'z')
+      /* 8.a */
+      if (rad < 2 || rad > 36)
       {
-        utf8_string_buff[i] = (lit_utf8_byte_t) (utf8_string_buff[i] - 'a' + 10);
+        ecma_number_t *ret_num_p = ecma_alloc_number ();
+        *ret_num_p = ecma_number_make_nan ();
+        ret_value = ecma_make_normal_completion_value (ecma_make_number_value (ret_num_p));
       }
-      else if (utf8_string_buff[i] >= 'A' && utf8_string_buff[i] <= 'Z')
+      /* 8.b */
+      else if (rad != 16)
       {
-        utf8_string_buff[i] = (lit_utf8_byte_t) (utf8_string_buff[i] - 'A' + 10);
+        strip_prefix = false;
       }
-      else if (lit_char_is_decimal_digit (utf8_string_buff[i]))
+    }
+    /* 9. */
+    else
+    {
+      rad = 10;
+    }
+
+    if (ecma_is_completion_value_empty (ret_value))
+    {
+      /* 10. */
+      if (strip_prefix)
       {
-        utf8_string_buff[i] = (lit_utf8_byte_t) (utf8_string_buff[i] - '0');
+        if (end - start >= 2
+            && utf8_string_buff[start] == '0'
+            && (utf8_string_buff[start + 1] == 'x' || utf8_string_buff[start + 1] == 'X'))
+        {
+          start += 2;
+
+          rad = 16;
+        }
       }
-      else
+
+      /* 11. Check if characters are in [0, Radix - 1]. We also convert them to number values in the process. */
+      for (lit_utf8_size_t i = start; i < end; i++)
       {
-        /* Not a valid number char, set value to radix so it fails to pass as a valid character. */
-        utf8_string_buff[i] = (lit_utf8_byte_t) rad;
+        if ((utf8_string_buff[i]) >= 'a' && utf8_string_buff[i] <= 'z')
```

Comment:
@zherczeg, I don't think it is wise to replace the with the equivalent `LIT_` define, as those are `ecma_char_t` types (`unit16_t`), but here we use `lit_utf8_byte_t` (`unit_8_t`). Should I really replace the constants?